### PR TITLE
Disable redux devtools support in prod mode

### LINKED ISSUE #1430: Redux DevTools can degradate heavily the overall platform performance
It is known that Redux Dev Tools can cause unpredictable performance issues as described [here](https://github.com/DSpace/dspace-angular/issues/1330).

At 4Science, we came across a very similar problem related to the MyDspace and we could only solve it by disabling the Redux DevTools in production.
For this particular customization (but not so uncommon in our opinion), a lot of information are embedded during the MyDspace search to properly display the search result items in an efficient way. So a lot of stuff has to be stored quickly in the Cache Store, but this not happen in time. The result is that every search result item causes singular rest requests to fetch information that should be easily retrieavable from the store.
Moreover (and as a consequence) a lot of UI elements result freezed during the Cache indexing.

Our testing proves that disabling the extension solves the problem and the user experience turns to be as smooth as usual.

Since we cannot make assumption over the presence of the Redux Dev Tools on the client side, our suggestion is to disable it by default in production environment.

### PR DESCRIPTION
**Pull request by alexandrevryghem**
## References
* Fixes #1430

## Description
Disabled support for redux devtools in production mode. There was no setting in `StoreDevtoolsModule.instrument` to disable it using the `environment.production` (followed [this article](https://ngohungphuc.wordpress.com/2020/02/26/turn-off-ngrxs-store-dev-tool-in-prod-mode/) to fix it). Other `dev-only` or 'prod-only` can now also be added to `src/config/store/devtools.ts` and `src/config/store/devtools.prod.ts`.

## Instructions for Reviewers
* Install the redux devtools plugin: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd
* Verify it is no longer supported (The icon should no longer light up, indicating it is not recording redux actions by default) when in prod mode (yrsp)
* Verify it is still supported in dev mode (yrsd)

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by atarix83 — APPROVED** [review]
thanks @alexandrevryghem for this PR

I've tested and the store is disabled as expected in production mode

File: src/config/store/devtools.prod.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const StoreDevModules = [];
```
Comment (tdonohue) [inline src/config/store/devtools.prod.ts:1]: Could we add a comment here to say:  "This disabled DevTools in Production mode, overriding the defaults in `devtools.ts`"

I want to be sure we are describing these new configs, as they are not very straightforward on their own.

File: src/config/store/devtools.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { environment } from '../../environments/environment';
+
+export const StoreDevModules = [
```
Comment (tdonohue) [inline src/config/store/devtools.ts:4]: Could we add a comment here to say: "This enables Redux DevTools by default.  Note: this is overridden in production by devtools.prod.ts".

Again, I'm not sure these settings will be easy to understand, so at least some basic comments would be worthwhile.

File: src/config/store/devtools.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  StoreDevtoolsModule.instrument({
+    maxAge: 1000,
+    logOnly: environment.production,
```
Comment (tdonohue) [inline src/config/store/devtools.ts:7]: Won't this always be false?  DevTools aren't enabled in production, so `environment.production` should always return false.   Should we make this an explicit `false` just to make that clearer?

**Review by tdonohue — COMMENTED** [review]
@alexandrevryghem : Overall, this looks great!  However, I have some minor requests before we merge this PR.  I'd appreciate it if we add comments for the new config files, as I'm worried they will not be self explanatory to other developers.  Other minor suggestions below.

Beyond that, I'm basically a +1, but I'd appreciate it if we can apply this minor feedback before merging.

**Pull request merged**